pkg/composableschemadsl/generator/generator_test.go

```diff
@@ -360,6 +359,32 @@ definition foos/document {
 }`,
 			`definition document {
 	permission first = rela->relb + relc.any(reld) + rele.all(relf)
+}`,
+		},
+		{
+			"expiration trait",
+			`use expiration
+			
+			definition document{
+				relation viewer: user with expiration
+				relation editor: user with somecaveat and expiration
+		}`,
+			`use expiration
+
+definition document {
+	relation viewer: user with expiration
+	relation editor: user with somecaveat and expiration
+}`,
+		},
```

Comment:
Note that this does not port in the expiration caveat tests, as we treat `expiration` as a reserved keyword in composable schema and treat it only as a trait.